Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
+        }
+      }
+      catch (const MolSanitizeException &){
+        std::cerr << "Molecule could not be sanitized" << std::endl;
+      }
+    }
+  }
+  else{
+    mol=SmilesToMol(txt,0,sanitize,replacements);
+  }
+  return mol;
+}
+
+
+
+
 }  // end of namespace DaylightParserUtils
 
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
-    bool useSmiles) {
+    bool useSmiles, bool sanitize) {
```

Comment:
I think my point is smiles should always sanitize, otherwise some matching features won't work and smarts never should.

----
Brian Kelley

> On Mar 31, 2016, at 8:26 AM, Greg Landrum <notifications@github.com> wrote:
> 
> In Code/GraphMol/ChemReactions/DaylightParser.cpp:
> 
> >  }  // end of namespace DaylightParserUtils
> >  
> >  ChemicalReaction *RxnSmartsToChemicalReaction(
> >      const std::string &text, std::map<std::string, std::string> *replacements,
> > -    bool useSmiles) {
> > +    bool useSmiles, bool sanitize) {
> Having "sanitize" as an option together with "useSmiles" is, to me, perfectly reasonable.
> 
> If the user calls this with sanitize=True and useSmiles=False, they kind of deserve whatever they get.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>